server/src/main/java/org/opensearch/common/cache/tier/CacheTierPolicy.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.common.cache.tier;
+
+import org.opensearch.core.common.bytes.BytesReference;
+
+import java.io.IOException;
+
+/**
+ * An interface for policies that inspect data of type T to decide whether they are admitted into a cache tier.
+ */
+public interface CacheTierPolicy<T> {
+    /**
+     * Determines whether this policy allows the data into its cache tier.
+     * @param data The data to check
+     * @return true if accepted, otherwise false
+     */
+    boolean checkData(T data);
+
+    /**
+     * Checks whether the policy is currently being enforced.
+     */
+    boolean isActive();
+
+    /**
+     * Starts enforcing the policy if it is not already being enforced.
+     */
+    void activate();
+
+    /**
+     * Stops enforcing the policy. The policy will allow all input.
+     */
+    void deactivate();
```

Comment:
Why do we need this? If we set a cachingPolicy in TieredService, then it should be always enforced?